arch/riscv/core/pmp/core_pmp.c

```diff
@@ -582,24 +1064,18 @@ int arch_mem_domain_thread_remove(struct k_thread *thread)
 #endif /* CONFIG_USERSPACE */
 
 #ifdef CONFIG_PMP_STACK_GUARD
-
 void z_riscv_init_stack_guard(struct k_thread *thread)
 {
-	unsigned char index = 0U;
-	unsigned char *uchar_pmpcfg;
 	ulong_t stack_guard_addr;
-
-	uchar_pmpcfg = (unsigned char *) thread->arch.s_pmpcfg;
-
-	uchar_pmpcfg++;
+	struct riscv_pmp_region dynamic_regions[3]; /* Maximum region_num is 3 */
```

Comment:
In commit 4478fbaa0130464410dadf183553f0afbbf14cce you introduce the maximum dynamic region number as 4, then you change it to 3. Why is that the case?

Reply:
Commit c09593fa11555aaf8a9597c8364d9492ae3ec70a add static PMP regions, and the MPRV region (enable permission by default) should have lower priority than static regions.
So I manage MPRV region globally in `mprv_region` and remove it from dynamic PMP regions in `z_riscv_init_stack_guard()`